CorsixTH/Lua/languages/string_ids.lua

```diff
@@ -0,0 +1,414 @@
+AdviserStringIds = {
+    boiler_issue = {
+        maximum_heat                  = 1,
+        minimum_heat                  = 2,
+        resolved                      = 3,
+    },
+
+    build_advice = {
+        blueprint_invalid             = 4,
+        blueprint_would_block         = 5,
+        door_not_reachable            = 6,
+        placing_object_blocks_door    = 7,
+    },
+
+    cheats = {
+        th_cheat                      = 8,
+        roujin_on_cheat               = 9,
+        roujin_off_cheat              = 10,
+        norest_on_cheat               = 11,
+        norest_off_cheat              = 12,
+    },
```

Comment:
You have along list of consecutive numbers. The collection strings however is not static. New strings are added, old strings are retired.

Imagine we add an `advisor_cheat`. What number does it get?

(And no, renumbering all lines below this is not an option. It leads to large file changes that are not readable.)

----
As a suggestion, why not use the string name itself? (That is, why must it be translated to a number at all?)

Reply:
As a second reason why these tables with numbers don't work: When writing a savegame, all data in the running program is stored.

When you load a savegame (eg one year later), all data is restored to its original spot. If you store these numbers anywhere (eg inside your window class), you thus get numbers of strings as they were one year ago.
Obviously, those numbers are wrong since in the mean time, strings have been added and have been removed. A number that you store now for (by example) `advisor_cheat` may be used for the completely unrelated string `mutiplayer_disconnected` one year later.